airflow/providers/amazon/aws/hooks/neptune.py

```diff
@@ -0,0 +1,113 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Interact with AWS Neptune."""
+from __future__ import annotations
+
+import time
+from typing import Callable
+
+from airflow.exceptions import AirflowException, AirflowNotFoundException
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+
+
+class NeptuneHook(AwsBaseHook):
+    """
+    Interact with AWS Neptune using proper client from the boto3 library.
+
+    Hook attribute `conn` has all methods that listed in documentation
+
+    .. seealso::
+        - https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/neptune.html
+        - https://docs.aws.amazon.com/neptune/index.html
+
+    Additional arguments (such as ``aws_conn_id`` or ``region_name``) may be specified and
+    are passed down to the underlying AwsBaseHook.
+
+    .. seealso::
+        :class:`~airflow.providers.amazon.aws.hooks.base_aws.AwsGenericHook`
+
+    :param aws_conn_id: The Airflow connection used for AWS credentials.
+    """
```

Comment:
Please define Hook docsting by the same way it is implemented in other boto3-hooks. See example

https://github.com/apache/airflow/blob/b314db9880ca9936cbe82e5527bf80d34d3d7861/airflow/providers/amazon/aws/hooks/ecs.py#L89-L101


You could check in [`main` branch documentation](http://apache-airflow-docs.s3-website.eu-central-1.amazonaws.com/docs/apache-airflow-providers-amazon/latest/_api/airflow/providers/amazon/aws/hooks/ecs/index.html#airflow.providers.amazon.aws.hooks.ecs.EcsHook) how it looks like.

Reply:
How can i check how my current doc string looks like in web UI ? Any chance ?

Reply:
You could build it in [Breeze](https://github.com/apache/airflow/blob/main/BREEZE.rst#building-the-documentation), be aware build entire documentation required a lot of  time, so make sure you only build for Amazon

Reply:
> 

The command for that is `breeze build-docs --package-filter apache-airflow-providers-amazon` and it does save a lot of build time. :+1: